subprojects/build-cache/src/main/java/org/gradle/caching/internal/origin/OriginMetadataFactory.java

```diff
@@ -70,14 +69,22 @@ public void execute(OutputStream outputStream) throws IOException {
                 properties.setProperty(EXECUTION_TIME_KEY, Long.toString(elapsedTime));
                 properties.setProperty(ROOT_PATH_KEY, rootDir.getAbsolutePath());
                 properties.setProperty(OPERATING_SYSTEM_KEY, operatingSystem);
-                properties.setProperty(HOST_NAME_KEY, inetAddressFactory.getHostname());
+                properties.setProperty(HOST_NAME_KEY, determineHostName());
                 properties.setProperty(USER_NAME_KEY, userName);
                 additionalProperties.accept(properties);
                 properties.store(outputStream, "Generated origin information");
             }
         };
     }
 
+    private static String determineHostName() {
+        try {
+            return InetAddress.getLocalHost().getHostName();
```

Comment:
Note that we won't obtain the correct host name as often as we did with `InetAddressFactory`, since `InetAddressFactory` contains some workarounds for different situations. It won't for example work in this case: https://github.com/gradle/gradle/pull/644

Though I don't think this is a big problem.

Reply:
Agreed. As I don't remember anyone ever looking at the host name in a cached artifact, I think this should be okay.